middlewares/cors.go

```diff
@@ -0,0 +1,139 @@
+package middlewares
+
+import (
+	"github.com/savsgio/atreugo/v10"
+	"github.com/savsgio/go-logger"
+	"github.com/valyala/fasthttp"
+
+	"strings"
+)
+
+// Create new properties for CORS.
+type CorsOptions struct {
+	// given origin or origins can be shared with response
+	AllowedOrigins []string
+
+	// which headers can be used during the actual request
+	AllowedHeaders []string
+
+	// seconds for caching the preflight result
+	AllowMaxAge int
+
+	// specify the method or methods allowed to resource
+	AllowedMethods []string
+
+	// which header names can be exposed as part of the response
+	ExposedHeaders []string
+
+	// whether to expose the response to frontend code
+	AllowCredentials bool
+}
+
+type CorsHandler struct {
+	allowedHeadersAll bool
+	allowedOrigins    []string
+	allowedHeaders    []string
+	allowedMethods    []string
+	exposedHeaders    []string
+}
+
+var defaultOptions = CorsOptions{
+	AllowedOrigins: []string{"*"},
+	AllowedMethods: []string{"GET", "POST"},
+	AllowedHeaders: []string{"Origin", "Accept", "Content-Type"},
+}
+
+func DefaultCors() *CorsHandler {
+	return NewCorsMiddleware(defaultOptions)
+}
+
+func NewCorsMiddleware(options CorsOptions) *CorsHandler {
+	cors := CorsHandler{
```

Comment:
Not duplicate the fields, just store options in `CorsHandler`

```go
cors := CorsHandler{
   options: options
}
```

Reply:
Hi @savsgio How to improve it?
The others of this file should have been resolved,

And I tested out that this file unavailable across domains, no reason found.

Thanks.

Reply:
And please, not open more PR about same theme!

Reply:
`And I tested out that this file unavailable across domains, no reason found.`

What do you mean?

Reply:
> And please, not open more PR about same theme!

OK, I'm sorry about 3 PRs.

> What do you mean?

I fix first, then I will check this again.
Thanks.